dependencies.yaml

```diff
@@ -258,13 +258,17 @@ dependencies:
           - *cmake_ver
           - cython>=3.0.3
           - *ninja
-          - &numpy numpy>=1.21
           # Hard pin the patch version used during the build. This must be kept
           # in sync with the version pinned in get_arrow.cmake.
           - pyarrow==14.0.2.*
       - output_types: conda
         packages:
           - scikit-build-core>=0.7.0
+      - output_types: pyproject
+        packages:
+          # Hard pin the patch version used during the build.
+          # Sync with conda build constraint & wheel run constraint.
+          - numpy==1.23.*
```

Comment:
This is the same NumPy pin the conda packages use to build. It is now used here to align the wheel packages